src/dreammaker/config.rs

```diff
@@ -0,0 +1,146 @@
+// various config options
+
+use serde::{Deserialize, Deserializer};
+use std::fs::File;
+use std::io::Read;
+use crate::error::Severity;
+
+#[derive(Debug, Clone, Copy)]
+pub enum WarningLevel {
+    Error = 1,
+    Warning = 2,
+    Info = 3,
+    Hint = 4,
+    Disabled = 5,
+    Unset = 6,
+}
+
+impl Default for WarningLevel {
+    fn default() -> WarningLevel {
+        WarningLevel::Unset
+    }
+}
+
+impl<'de> Deserialize<'de> for WarningLevel {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?.to_lowercase();
+        let warninglevel = match s.as_str() {
+            "error" | "errors" => Ok(WarningLevel::Error),
+            "warning" | "warnings" => Ok(WarningLevel::Warning),
+            "info" | "infos" => Ok(WarningLevel::Info),
+            "hint" | "hints" => Ok(WarningLevel::Hint),
+            "false" | "off" => Ok(WarningLevel::Disabled),
+            _other => Ok(WarningLevel::Unset),
+        };
+        warninglevel
+    }
+}
+
+impl From<WarningLevel> for Severity {
+    fn from(warninglevel: WarningLevel) -> Self {

```

Comment:
This `From` doesn't seem to be a complete conversion, perhaps the `match` where it's used should simply be made complete instead.